### DISCUSSION TITLE
For booster-http consider adding transparent gzip support

### DISCUSSION THREAD

**Discussion by ribasushi** [filecoin-project/boost#826] [Ideas]
https://pkg.go.dev/github.com/NYTimes/gziphandler#readme-usage

**Comment by dirkmc** [comment]
Seems like a good idea 👍 
My only concern would be how CPU intensive compression would be, as the pieces are large files. @ribasushi do you have a sense of what the trade-off looks like between how well a piece would be compressed vs how much CPU resources it takes to compress the data?

**Comment by ribasushi** [comment]
Gzip is effectively a SIMD-optimized process these days, the overhead on a modern ( last ~5y ) CPU is a rounding error.

**Comment by dirkmc** [comment]
Great, thanks 👍 
I added it as an issue: https://github.com/filecoin-project/boost/issues/830

**Comment by nonsense** [comment]
Locking discussion as this has been implemented.
